Source/DafnyLanguageServer/Plugins/DafnyCodeActionHelpers.cs

```diff
@@ -0,0 +1,193 @@
+﻿using System;
+using Microsoft.Boogie;
+using Microsoft.Dafny.LanguageServer.Language;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using Range = OmniSharp.Extensions.LanguageServer.Protocol.Models.Range;
+
+namespace Microsoft.Dafny.LanguageServer.Plugins;
+
+/// <summary>
+/// Helpers for plugins defining a DafnyCodeActionProvider
+/// </summary>
+public static class DafnyCodeActionHelpers {
+  /// <summary>
+  /// Given an LSP range and some text, extract the corresponding substring
+  /// </summary>
+  /// <param name="range">The range to extract</param>
+  /// <param name="text">The document</param>
+  /// <returns>The substring of the document in the range</returns>
+  public static string Extract(Range range, string text) {
```

Comment:
I don't think this linear time complexity method should exist. We can add an `Extract(Range range)` to `DocumentTextBuffer` though.